airbyte-persistence/job-persistence/src/main/java/io/airbyte/persistence/job/WorkspaceHelper.java

```diff
@@ -0,0 +1,204 @@
+  }
+
+  public UUID getWorkspaceForSourceIdIgnoreExceptions(final UUID sourceId) {
+    return swallowExecutionException(() -> getWorkspaceForSourceId(sourceId));
+  }
+
+  // DESTINATION ID
+  public UUID getWorkspaceForDestinationId(final UUID destinationId) throws JsonValidationException, ConfigNotFoundException {
+    return handleCacheExceptions(() -> destinationToWorkspaceCache.get(destinationId));
+  }
+
+  public UUID getWorkspaceForDestinationIdIgnoreExceptions(final UUID destinationId) {
+    return swallowExecutionException(() -> destinationToWorkspaceCache.get(destinationId));
+  }
+
+  // JOB ID
+  public UUID getWorkspaceForJobId(final Long jobId) throws JsonValidationException, ConfigNotFoundException {
+    return handleCacheExceptions(() -> jobToWorkspaceCache.get(jobId));
+  }
+
+  public UUID getWorkspaceForJobIdIgnoreExceptions(final Long jobId) {
+    return swallowExecutionException(() -> jobToWorkspaceCache.get(jobId));
+  }
+
+  // CONNECTION ID
+  public UUID getWorkspaceForConnection(final UUID sourceId, final UUID destinationId) throws JsonValidationException, ConfigNotFoundException {
+    final UUID sourceWorkspace = getWorkspaceForSourceId(sourceId);
+    final UUID destinationWorkspace = getWorkspaceForDestinationId(destinationId);
+
+    Preconditions.checkArgument(Objects.equals(sourceWorkspace, destinationWorkspace), "Source and destination must be from the same workspace!");
+    return sourceWorkspace;
+  }
+
+  public UUID getWorkspaceForConnectionIgnoreExceptions(final UUID sourceId, final UUID destinationId) {
+    final UUID sourceWorkspace = getWorkspaceForSourceIdIgnoreExceptions(sourceId);
+    final UUID destinationWorkspace = getWorkspaceForDestinationIdIgnoreExceptions(destinationId);
+
+    Preconditions.checkArgument(Objects.equals(sourceWorkspace, destinationWorkspace), "Source and destination must be from the same workspace!");
+    return sourceWorkspace;
+  }
+
+  public UUID getWorkspaceForConnectionId(final UUID connectionId) throws JsonValidationException, ConfigNotFoundException {
+    return handleCacheExceptions(() -> connectionToWorkspaceCache.get(connectionId));
+  }
+
+  public UUID getWorkspaceForConnectionIdIgnoreExceptions(final UUID connectionId) {
+    return swallowExecutionException(() -> connectionToWorkspaceCache.get(connectionId));
+  }
+
+  // OPERATION ID
+  public UUID getWorkspaceForOperationId(final UUID operationId) throws JsonValidationException, ConfigNotFoundException {
+    return handleCacheExceptions(() -> operationToWorkspaceCache.get(operationId));
+  }
+
+  public UUID getWorkspaceForOperationIdIgnoreExceptions(final UUID operationId) {
+    return swallowExecutionException(() -> operationToWorkspaceCache.get(operationId));
+  }
+
+  private static UUID handleCacheExceptions(final CheckedSupplier<UUID, ExecutionException> supplier)
+      throws ConfigNotFoundException, JsonValidationException {
+    try {
+      return supplier.get();
```

Comment:
Can we change the "get" method to throw the right exception instead of doing the checks on the cause?